test_util/provide_aws.py

```diff
@@ -0,0 +1,215 @@
+#!/usr/bin/env python3
+import logging
+import os
+from abc import ABCMeta, abstractmethod
+
+import boto3
+import botocore
+import retrying
+
+
+LOGGING_FORMAT = '[%(asctime)s|%(name)s|%(levelname)s]: %(message)s'
+logging.basicConfig(format=LOGGING_FORMAT, level=logging.DEBUG)
+logger = logging.getLogger(__name__)
+
+AWS_VERIFY = True
+
+
+class AwsProvision(metaclass=ABCMeta):
+    """Class for utilities shared between VPC and CF provisioning
+    """
+
+    @abstractmethod
+    def create(self):
+        pass
+
+    @abstractmethod
+    def delete(self):
+        pass
+
+    def __init__(self, region):
+        self.stack = None
+        self.stack_name = None
+        self.region = region
+        self.session = boto3.session.Session(
+            aws_access_key_id=os.environ['AWS_ACCESS_KEY_ID'],
+            aws_secret_access_key=os.environ['AWS_SECRET_ACCESS_KEY'])
+        self.ec2_client = self.session.client(
+            service_name='ec2',
+            region_name=self.region,
+            verify=AWS_VERIFY)
+
+    def set_expiration(self, expiration_string):
```

Comment:
Anything created by AWS CloudFormation falls under the jurisdiction of of CCM for cleanup rather than the owner + expiration CloudCleaner does.

Need to use: https://ccm.mesosphere.com/api-docs/#!/cluster/Cluster_List_POST to set / extend the expiration.